packages/pluggableWidgets/column-chart-native/src/components/ColumnChart.tsx

```diff
@@ -0,0 +1,367 @@
+                                            orientation={"bottom"}
+                                            style={mapToAxisStyle(style.grid, style.xAxis)}
+                                            {...(firstSeries?.xFormatter
+                                                ? { tickFormat: firstSeries.xFormatter }
+                                                : undefined)}
+                                        />
+                                        <VictoryAxis
+                                            style={mapToAxisStyle(style.grid, style.yAxis)}
+                                            orientation={"left"}
+                                            dependentAxis
+                                            {...(firstSeries?.yFormatter
+                                                ? { tickFormat: firstSeries.yFormatter }
+                                                : undefined)}
+                                        />
+                                        {groupedOrStacked}
+                                    </VictoryChart>
+                                ) : null}
+                            </View>
+                            {axisLabelStyles.extractedXAxisLabelStyle.relativePositionGrid === "right"
+                                ? xAxisLabelComponent
+                                : null}
+                        </View>
+                        {axisLabelStyles.extractedXAxisLabelStyle.relativePositionGrid === "bottom"
+                            ? xAxisLabelComponent
+                            : null}
+                    </View>
+                    {showLegend ? (
+                        <Legend style={style.legend} series={sortedSeries} seriesColors={normalizedColumnColors} />
+                    ) : null}
+                </View>
+            )}
+        </View>
+    );
+}
+
+type DataTypeResult = { x: string; y: string };
+
+function getDataTypes(series: ColumnChartSeries[]): DataTypeResult | Error | undefined {
+    let dataTypes: DataTypeResult | undefined;
+
+    for (const _series of series) {
+        const { dataPoints } = _series;
+
+        if (dataPoints.length) {
+            const { x, y } = dataPoints[0];
+            const xDataType = typeof x;
+            const yDataType = typeof y;
+
+            if (!dataTypes) {
+                dataTypes = { x: xDataType, y: yDataType };
+            } else if (dataTypes && (dataTypes.x !== xDataType || dataTypes.y !== yDataType)) {
+                return new Error("Data types of data points belonging to different series aren't equal");
+            }
+        }
+    }
+
+    return dataTypes;
+}
+
+const getScale = (dataTypesResult: string): "linear" | "time" | undefined => {
+    if (dataTypesResult === "number" || dataTypesResult === "string") {
+        return "linear";
+    } else if (dataTypesResult === "object") {
+        return "time";
+    }
+
+    return undefined;
```

Comment:
I would personally go for a `switch` here, but I'll leave that up to you :).

Reply:
done